### PR TITLE
[native] Support reading Iceberg V2 tables with Position deletes

### PR DESCRIPTION
**Pull request by imjalpreet**
## Description

This PR adds support for reading Iceberg V2 tables with Position Deletes. An unsupported exception will be thrown if the table contains Equality Deletes

## Motivation and Context

https://github.com/facebookincubator/velox/pull/7847

## Impact

With this change, users will be able to read a subset of Iceberg V2 Tables, i.e. tables with position deletes. 

## Test Plan

Tested manually by creating Iceberg V2 tables with multiple Position Deletes.

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

Prestissimo (Native Execution) Changes
* Add support to read Iceberg V2 tables with Position Deletes 

```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxSplit.cpp [hunk 1]
```diff
@@ -114,14 +126,55 @@ velox::exec::Split toVeloxSplit(
...
 
+    std::unordered_map<std::string, std::string> customSplitInfo;
+    customSplitInfo["table_format"] = "hive-iceberg";
+
+    std::vector<velox::connector::hive::iceberg::IcebergDeleteFile> deletes;
+    deletes.reserve(icebergSplit->deletes.size());
+    for (const auto& deleteFile : icebergSplit->deletes) {
+      std::vector<int32_t> equalityFieldIds;
+      equalityFieldIds.reserve(deleteFile.equalityFieldIds.size());
+      for (auto field : deleteFile.equalityFieldIds) {
```
Comment (yingsu00) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxSplit.cpp:137]: Can you try to just use the copy constructor std::vector(const std::vector& another)? List is the same as std::vector<T>

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxSplit.cpp [hunk 2]
```diff
@@ -114,14 +126,55 @@ velox::exec::Split toVeloxSplit(
...
+        equalityFieldIds.emplace_back(field);
+      }
+
+      std::unordered_map<int32_t, std::string> lowerBounds;
```
Comment (yingsu00) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxSplit.cpp:141]: Can you try `template< class InputIt > unordered_map( InputIt first, InputIt last)`?

**Comment by yingsu00** [comment]
@imjalpreet Have you tried the copy constructors yet? Do they work?

**Comment by imjalpreet** [comment]
> Have you tried the copy constructors yet? Do they work?
@yingsu00 sorry I forgot to push my commit, I thought I had already done that. Yes, I had made the changes and those worked.

**Review by yingsu00 — APPROVED** [review]

**Comment by yingsu00** [comment]
@imjalpreet There are test failures
```
org.apache.iceberg.exceptions.NotFoundException: Failed to open input stream for file: file:/tmp/PrestoTest1442477645755568799/iceberg_data/catalog/test_metadata_schema/iceberg_table1/metadata/00000-ff582f52-419c-493e-90f2-f1ea493599cb.metadata.json
```
Can you please take a look? Thanks!

**Comment by imjalpreet** [comment]
>  There are test failures

Unrelated failures since we haven't made any changes to the Java code. Most probably intermittent, I have re-run the test suite to verify.

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxSplit.cpp (hunk 2 above)
Comment (imjalpreet) [reply presto-native-execution/presto_cpp/main/types/PrestoToVeloxSplit.cpp]: Done

**Pull request merged**